src/test/test_utils.rs

```diff
@@ -445,3 +479,101 @@ where
         .map(|x| std::io::IoSliceMut::new(x.as_mut()))
         .collect()
 }
+
+#[derive(Debug, Copy, Clone, Eq, Ord, PartialEq, PartialOrd)]
+pub enum VerifyOrder {
+    First,
+    Second,
+    Third,
+    Fourth,
+    Fifth,
+    Sixth,
+}
+
+#[derive(Debug, Copy, Clone, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Verify {
+    pub order: VerifyOrder,
+    pub rv: Option<libc::c_int>,
+    pub errno: Option<libc::c_int>,
+}
```

Comment:
(Optional) I don't completely understand this code, but instead of having an abstract "First", "Second", etc, maybe it would be nicer if `Verify` was generic over `T: Debug + Copy + Clone + Eq + Ord + PartialEq + PartialOrd` and used `pub order: T` instead of `pub order: VerifyOrder`. That way the calling code could call `Verify::new()` with its own enum (ex: `enum ClockVerifyOrder { ClockId, Request, Flags }`, or even just `u8`) and use its own ordering instead. But maybe having just "first", "second", etc is clearer. Either way is fine with me, just with some more documentation.